pkg/testutil/framework.go

```diff
@@ -307,6 +315,182 @@ func (f *Framework) SetupServiceCluster(ctx context.Context, cl *RecordingClient
+				Namespace: namespaceName,
+			},
+		},
+		// Role
+		&rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-leader-election-role", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		&apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "accounts.catalog.kubecarrier.io",
+			},
+		},
+	})
+}
+
+func (f *Framework) E2EOperatorCheck(
+	ctx context.Context,
+	t *testing.T,
+	serviceClient *RecordingClient,
+	serviceScheme *runtime.Scheme,
+) {
+	namePrefix := "e2e-operator"
+	namespaceName := "kubecarrier-e2e-operator"
+
+	f.componentCheck(ctx, serviceClient, serviceScheme, t, namePrefix, []runtime.Object{
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-manager", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// ClusterRoleBinding
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-rolebinding", namePrefix),
+			},
+		},
+		// ClusterRole
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-role", namePrefix),
+			},
+		},
+		&apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "dbs.fake.kubecarrier.io",
+			},
+		},
+	})
+}
+
+func (f *Framework) KubeCarrierCheck(
+	ctx context.Context,
+	t *testing.T,
+	managementClient *RecordingClient,
+	managementScheme *runtime.Scheme,
+) {
+	namePrefix := constants.KubeCarrierDefaultName + "-manager"
+	namespaceName := constants.KubeCarrierDefaultNamespace
+
+	f.componentCheck(ctx, managementClient, managementScheme, t, namePrefix, []runtime.Object{
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-controller-manager", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// Webhook Service
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-webhook-service", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// ClusterRoleBinding
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-rolebinding", namePrefix),
+			},
+		},
+		// ClusterRole
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-role", namePrefix),
+			},
+		},
+		// RoleBinding
+		&rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-leader-election-rolebinding", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// Role
+		&rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-leader-election-role", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+	})
+}
+
+func (f *Framework) componentCheck(
```

Comment:
why does this function have Framework as pointer receiver when it's unused? It could very well stand on its own.

Reply:
There are serval functions in the `framework` package which have it as pointer receiver and don't use it, I wanted to have that so it can keep in the context of `framework`.